flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocatorTest.java

```diff
@@ -0,0 +1,575 @@
+		}
+
+		static Builder newBuilder() {
+			return new Builder();
+		}
+
+		private static class Builder {
+			private int[] groups = { 2, 1 }; // 2 executions in the first group, 1 in the second etc
+			private List<ResourceProfile> resourceProfiles;
+			private boolean completePhysicalSlotFutureManually;
+			private boolean completeSlotProfileFutureManually;
+			private boolean slotWillBeOccupiedIndefinitely;
+
+			private Builder setGroups(int... groups) {
+				int reqNumber = IntStream.of(groups).sum();
+				List<ResourceProfile> resourceProfiles = Collections.nCopies(reqNumber, ResourceProfile.UNKNOWN);
+				return setGroupsWithResourceProfiles(resourceProfiles, groups);
+			}
+
+			private Builder setGroupsWithResourceProfiles(List<ResourceProfile> resourceProfiles, int... groups) {
+				Preconditions.checkArgument(resourceProfiles.size() == IntStream.of(groups).sum());
+				this.resourceProfiles = resourceProfiles;
+				this.groups = groups;
+				return this;
+			}
+
+			private Builder completePhysicalSlotFutureManually() {
+				completePhysicalSlotFutureManually(true);
+				return this;
+			}
+
+			private Builder completePhysicalSlotFutureManually(boolean value) {
+				this.completePhysicalSlotFutureManually = value;
+				return this;
+			}
+
+			private Builder completeSlotProfileFutureManually() {
+				this.completeSlotProfileFutureManually = true;
+				return this;
+			}
+
+			private Builder setSlotWillBeOccupiedIndefinitely(boolean slotWillBeOccupiedIndefinitely) {
+				this.slotWillBeOccupiedIndefinitely = slotWillBeOccupiedIndefinitely;
+				return this;
+			}
+
+			private AllocationContext build() {
+				List<ExecutionVertexSchedulingRequirements> requirements = createSchedulingRequirements();
+				TestingPhysicalSlotProvider slotProvider = new TestingPhysicalSlotProvider(completePhysicalSlotFutureManually);
+				TestingSharedSlotProfileRetrieverFactory sharedSlotProfileRetrieverFactory =
+					new TestingSharedSlotProfileRetrieverFactory(completeSlotProfileFutureManually);
+				TestingSlotSharingStrategy slotSharingStrategy = TestingSlotSharingStrategy.createWithGroups(getReqIds(requirements), groups);
+				SlotSharingExecutionSlotAllocator allocator = new SlotSharingExecutionSlotAllocator(
+					slotProvider,
+					slotWillBeOccupiedIndefinitely,
+					slotSharingStrategy,
+					sharedSlotProfileRetrieverFactory);
+				return new AllocationContext(
+					requirements,
+					slotProvider,
+					slotSharingStrategy,
+					allocator,
+					sharedSlotProfileRetrieverFactory);
+			}
+
+			private List<ExecutionVertexSchedulingRequirements> createSchedulingRequirements() {
+				return resourceProfiles
+					.stream()
+					.map(resourceProfile ->
+						new ExecutionVertexSchedulingRequirements
+							.Builder()
+							.withExecutionVertexId(new ExecutionVertexID(new JobVertexID(), 0))
+							.withTaskResourceProfile(resourceProfile)
```

Comment:
Both `taskResourceProfile` and `physicalSlotResourceProfile` of `ExecutionVertexSchedulingRequirements` are not used in `SlotSharingExecutionSlotAllocator`. I think we do not need to set them and therefore no need to create the `resourceProfiles`.